addon/mixins/attachable.js

```diff
@@ -75,7 +75,7 @@ export default Ember.Mixin.create({
     }
   },
   _rootKey: function() {
-    return Ember.String.underscore(Ember.String.decamelize(this.constructor.typeKey));
+    return Ember.String.underscore(this._modelName);
```

Comment:
@mbutsko I think you forgot to add back the `Ember.String.decamelize` call here.